notifier/src/main/kotlin/Notifier.kt

```diff
@@ -43,15 +43,20 @@ class Notifier(
 ) : ScriptRunner() {
     private val customProperties = buildMap {
         put("ortResult", ortResult)
-
-        config.mail?.let { put("mailClient", MailNotifier(it)) }
-        config.jira?.let { put("jiraClient", JiraNotifier(it)) }
-
+        put("mailClient", config.mail?.let { MailNotifier(it) })
+        put("jiraClient", config.jira?.let { JiraNotifier(it) })
         put("resolutionProvider", resolutionProvider)
     }
 
+    private val customPropertyTypes = mapOf(
+        "ortResult" to KotlinType(OrtResult::class),
+        "mailClient" to KotlinType(MailNotifier::class, isNullable = true),
+        "jiraClient" to KotlinType(JiraNotifier::class, isNullable = true),
+        "resolutionProvider" to KotlinType(ResolutionProvider::class)
+    )
+
     override val compConfig = createJvmCompilationConfigurationFromTemplate<NotificationsScriptTemplate> {
-        providedProperties(customProperties.mapValues { (_, v) -> KotlinType(v::class) })
+        providedProperties(customPropertyTypes)
```

Comment:
This could be inlined, like done [here](https://github.com/eclipse-apoapsis/ort-server/pull/278/files#diff-51ef293145d9ce26717e363cfa7da707dc7cb6792e54c2e93503c0f5c98e9cd4R80) (using a vararg of pairs). And I'd probably inline `customProperties` then as well.